read.go

```diff
@@ -269,10 +265,52 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 
 	debug.Logf("readNativeFrames:\nRows: %d\nCols:%d\nFrames::%d\nBitsAlloc:%d\nSamplesPerPixel:%d", MustGetInts(rows.Value)[0], MustGetInts(cols.Value)[0], nFrames, bitsAllocated, samplesPerPixel)
 
+	bytesAllocated := bitsAllocated / 8
+	bytesToRead = bytesAllocated * samplesPerPixel * pixelsPerFrame * nFrames
+	if bitsAllocated == 1 {
+		bytesToRead = pixelsPerFrame * samplesPerPixel / 8 * nFrames
+	}
+
+	skipOne := false
+	switch {
+	case uint32(bytesToRead) == vl: // we good nothing
+	case uint32(bytesToRead) == vl-1 && vl%2 == 0:
+		skipOne = true
+	case uint32(bytesToRead) == vl-1 && vl%2 != 0:
+		return nil, 0, fmt.Errorf("vl=%d: %w", vl, ErrorExpectedEvenLength)
+	default:
+		if !parsedData.opts.allowMismatchPixelDataLength {
+			return nil, 0, fmt.Errorf("expected_vl=%d actual_vl=%d %w", bytesToRead, vl, ErrorMismatchPixelDataLength)
+		}
+		data := make([]byte, vl)
+		_, err = io.ReadFull(d, data)
+		if err != nil {
+			return nil, 0, fmt.Errorf("read pixelData: %w", err)
+		}
+
+		f := frame.Frame{
+			VLMismatch: true,
+			EncapsulatedData: frame.EncapsulatedFrame{
+				Data: data,
+			},
+		}
+
+		if fc != nil {
+			fc <- &f
+		}
+		image := PixelDataInfo{
+			IsVLMismatch: true,
+		}
+		image.Frames = append(image.Frames, f)
```

Comment:
can we move this logic to a helper function, something like `makeErrorPixelData(data io.Reader, frameChannel, errToInsert error) *PixelDataInfo`. This is assuming we go with my frame.ParsingErr suggestion (or potentially ParsingErr could go on the PixelDataInfo outright, since I'm not sure it needs to be frame level). (readNativeFrames is already so complicated, probably best to factor out logic for readability).